Tasks/AzureCLIV2/azureclitask.ts

```diff
@@ -128,7 +128,7 @@ export class azureclitask {
             let escapedCliPassword = cliPassword.replace(/"/g, '\\"');
             tl.setSecret(escapedCliPassword.replace(/\\/g, '\"'));
             //login using svn
-            Utility.throwIfError(tl.execSync("az", `login --service-principal -u "${servicePrincipalId}" -p "${escapedCliPassword}" --tenant "${tenantId}"`), tl.loc("LoginFailed"));
+            Utility.throwIfError(tl.execSync("az", `login --service-principal -u "${servicePrincipalId}" -p "${escapedCliPassword}" --tenant "${tenantId}" --allow-no-subscriptions`), tl.loc("LoginFailed"));
```

Comment:
Won't line no 104 itself throw if the SPN is not tied to a subscription as the service connection will not have any subscription Id?

Reply:
May be we need to pass third parameter as false so that it does not throw

Reply:
Today, with the AzureCLI@2, it does seem to attempt to run the `az login` command. It throws an error at runtime if no subscription is assigned and the error suggests adding `--allow-no-subscriptions` to suppress the error. This change here is suppressing the error and allowing the task to successfully login even though it has not direct RBAC/IAM access to Azure resources.

Reply:
on this line, the optional flag is set to `true` so I do not think any error will be thrown.
https://github.com/microsoft/azure-pipelines-tasks/blob/bf7f4cdf248af83e68ab5b40db51588269764ed8/Tasks/AzureCLIV2/azureclitask.ts#L104

Reply:
to be clear, this is looking at the parameter, not the actual account.